### ISSUE TITLE
Add pre-flight check: test for 'docker-machine-driver-kvm' before downloading/starting

### ISSUE THREAD

**Issue by ctron** [minishift/minishift#966]
Downloading and unpacking minishift might result in the following error:

```
➜  minishift ./minishift start --metrics --memory  8192
Starting local OpenShift cluster using 'kvm' hypervisor...
Downloading ISO 'https://github.com/minishift/minishift-b2d-iso/releases/download/v1.0.2/minishift-b2d.iso'
 40.00 MiB / 40.00 MiB [======================================================================================================================================================================] 100.00% 0s
E0530 11:24:40.356849    3830 start.go:176] Error starting the VM: Driver "kvm" not found. Do you have the plugin binary "docker-machine-driver-kvm" accessible in your PATH?. Retrying.
E0530 11:24:40.358352    3830 start.go:176] Error starting the VM: Driver "kvm" not found. Do you have the plugin binary "docker-machine-driver-kvm" accessible in your PATH?. Retrying.
E0530 11:24:40.358434    3830 start.go:176] Error starting the VM: Driver "kvm" not found. Do you have the plugin binary "docker-machine-driver-kvm" accessible in your PATH?. Retrying.
Error starting the VM:  Driver "kvm" not found. Do you have the plugin binary "docker-machine-driver-kvm" accessible in your PATH?
Driver "kvm" not found. Do you have the plugin binary "docker-machine-driver-kvm" accessible in your PATH?
Driver "kvm" not found. Do you have the plugin binary "docker-machine-driver-kvm" accessible in your PATH?
```

A quick look at the documentation solves this issue, but it pre-flight check in minishift and maybe a pointer to the documentation on the command line would smooth that experience a bit.

**Comment by praveenkumar** [comment]
@ctron you mean test available driver binary before start process?

**Comment by ctron** [comment]
Yes, sorry I forgot to update the title before creating the issue 😉

**Comment by gbraad** [comment]
This is part of #485; pre-flight checks should happen before any action is taken. Download should never happen before the requirements are met.

**Comment by LalatenduMohanty** [comment]
@gbraad right this issue have the same line of thought as #485 but the exact preflight check is not covered in #485.

**Comment by gbraad** [comment]
although not covered per se, this task was created to deal with preflight checks: https://github.com/minishift/minishift/issues/485#issuecomment-284710930 You are free to suggest more...

**Comment by budhrg** [comment]
As we mentioned to add this as part of #485 here https://github.com/minishift/minishift/issues/485#issuecomment-284710930, I believe we can close this one.

WDYT @hferentschik @gbraad @LalatenduMohanty @praveenkumar .

**Comment by praveenkumar** [comment]
@budhrg here we just referenced this issue in our list so let it be open and we will close once we implement original.

**Comment by hferentschik** [comment]
Resolved as part of issue #485 and merged in pull request #1162
